tests/test_zernike.py

```diff
@@ -414,6 +418,214 @@ def finite_difference_gradient(Z, x, y):
+        coefDiff[:len(z1.coef)] -= c1*z1.coef
+        np.testing.assert_allclose(coefSum, (c1*z1 + c2*z2).coef)
+        np.testing.assert_allclose(coefDiff, -(c1*z1 - c2*z2).coef)
+
+        np.testing.assert_allclose(
+            c1*z1(x, y) + c2*z2(x, y),
+            (c1*z1 + c2*z2)(x, y)
+        )
+        np.testing.assert_allclose(
+            c1*z1(x, y) - c2*z2(x, y),
+            (c1*z1 - c2*z2)(x, y)
+        )
+        # Check that R_outer and R_inner are preserved
+        np.testing.assert_allclose(
+            (z1+z2).R_outer,
+            R_outer
+        )
+        np.testing.assert_allclose(
+            (z1+z2).R_inner,
+            R_inner
+        )
+
+    with np.testing.assert_raises(TypeError):
+        z1 + 3
+    with np.testing.assert_raises(TypeError):
+        z1 - 3
+    with np.testing.assert_raises(ValueError):
+        z1 + galsim.zernike.Zernike([0,1], R_outer=z1.R_outer*2)
+    with np.testing.assert_raises(ValueError):
+        z1 + galsim.zernike.Zernike([0,1], R_outer=z1.R_outer, R_inner=z1.R_inner*2)
+
+    # Commutative with integer coefficients
+    z1 = galsim.zernike.Zernike([0,1,2,3,4])
+    z2 = galsim.zernike.Zernike([1,2,3,4,5,6])
+    assert z1+z2 == z2+z1
+    assert (z2-z1) == z2 + -z1 == -(z1-z2)
+
+
+@timer
+def test_product():
+    """Test that __mul__ and __rmul__ work as expected.
+    """
+    u = galsim.UniformDeviate(57)
+    x = np.empty(100, dtype=float)
+    y = np.empty(100, dtype=float)
+    u.generate(x)
+    u.generate(y)
+
+    for _ in range(100):
+        n1 = int(u()*21)+3
+        n2 = int(u()*21)+3
+        R_outer = 1+0.1*u()
+        R_inner = 0.1*u()
+        a1 = np.empty(n1, dtype=float)
+        a2 = np.empty(n2, dtype=float)
+        u.generate(a1)
+        u.generate(a2)
+        z1 = galsim.zernike.Zernike(a1, R_outer=R_outer, R_inner=R_inner)
+        z2 = galsim.zernike.Zernike(a2, R_outer=R_outer, R_inner=R_inner)
+
+        np.testing.assert_allclose(
+            z1(x, y) * z2(x, y),
+            (z1 * z2)(x, y),
+            rtol=1e-6, atol=1e-6
+        )
+        np.testing.assert_allclose(
+            z1(x, y) * z2(x, y),
+            (z2 * z1)(x, y),
+            rtol=1e-6, atol=1e-6
+        )
+        # Check that R_outer and R_inner are preserved
+        np.testing.assert_allclose(
+            (z1*z2).R_outer,
+            R_outer
+        )
+        np.testing.assert_allclose(
+            (z1*z2).R_inner,
+            R_inner
+        )
+
+    with np.testing.assert_raises(TypeError):
+        z1 * galsim.Gaussian(fwhm=1)
+    with np.testing.assert_raises(ValueError):
+        z1 * galsim.zernike.Zernike([0,1], R_outer=z1.R_outer*2)
+    with np.testing.assert_raises(ValueError):
+        z1 * galsim.zernike.Zernike([0,1], R_outer=z1.R_outer, R_inner=z1.R_inner*2)
+
+    # Commutative with integer coefficients
+    z1 = galsim.zernike.Zernike([0,1,2,3,4,5])
+    z2 = galsim.zernike.Zernike([1,2,3,4,5,6])
+    assert z1*z2 == z2*z1
```

Comment:
I think multiplication by a scalar is only implicitly tested via `__sub__`.  Probably worth adding a real test of that.

Reply:
This is tested in `test_sum`, but I added it here too to make it easier to find.